ocaml/typing/types.mli

```diff
@@ -221,6 +217,26 @@ and abbrev_memo =
     This is only allowed when the real type is known.
 *)
 
+(** Jkinds classify types. *)
+(* CR layouts v2.8: Say more here. *)
+and jkind = type_expr Jkind_types.t
+
+(* jkind depends on types defined in this file, but some functions defined in jkind.ml
+   are required here. When jkind.ml is loaded, it fills these refs with the definitions
+   of the corresponding functions. *)
+
+(** INTERNAL USE ONLY
+    jkind.ml should fill this ref with the definition of Jkind.Sort.change_log *)
+val jkind_sort_change_log : (Jkind_types.Sort.change -> unit) ref ref
+
+(** INTERNAL USE ONLY
+    jkind.ml should fill this ref with the definition of Jkind.Sort.undo_change *)
+val jkind_sort_undo_change : (Jkind_types.Sort.change -> unit) ref
+
+(** INTERNAL USE ONLY
+    jkind.ml should fill this ref with the definition of Jkind.equal *)
+val jkind_equal : (jkind -> jkind -> bool) ref
```

Comment:
I think, given the comments I've written in types.ml, that this should be the one place a ref like this is needed.